### DISCUSSION TITLE
Direct SQL in a GridView.

### DISCUSSION THREAD

**Discussion by djbodya** [pimcore/pimcore#15634] [General]
Hello.
I was using previous version of pimcore but when i migrated to 11.0.5 i can not find "Direct SQL in a GridView". It is problem on my side or this feature does not available any more?

**Comment by NiklasBr** [comment]
It was removed for security reasons: https://github.com/pimcore/pimcore/pull/13395

**Reply by j-ts** [reply]
Hello, @NiklasBr. Thanks for pointing out! Do you know what was introduced instead, to select objects with images or other relations is not null? I see these _Filter by relation_ for images, for example, make it possible to filter objects without any image OR a specific image present. For my work it is very inconvenient, I used Direct SQL queries in GridView to check for errors in products, and now it's kind of impossible..

**Reply by NiklasBr** [reply]
There is  #15315 on the way, but I suspect it is not fully solving your use case?

Though you can filter by empty/none in the relations filter in Pimcore 11 today, applying this filter shows only the objects with empty Category.

![image](https://github.com/pimcore/pimcore/assets/279826/07020ee5-eca9-43d5-8f70-46c912876b70)

**Reply by j-ts** [reply]
Yes, it's a good feature, but not solving a problem if you need to select objects when a specific attribute is NOT empty. In image filter, you can't add them all to get the desired result, you know what I mean?

**Reply by NiklasBr** [reply]
Yes, looks like missing functionality.

**Reply by johanlindgren** [reply]
Maybe the SearchTerm field in advanced object search could be used (even if it's not exactly SQL)

**Reply by SamyMP** [reply]
Also might be related to #14666, where my need was to be able to filter on text values that are empty. I also used the SQL filter to achieve this but not possible anymore in Pimcore 11

**Comment by trnqeu** [comment]
I was using this feature a lot to search inside the grid for objects with or without a specific tag (or more than one tag). Is there any option to achieve the same result inside the grid? It's extremely important for us to edit objects based on a search by tag, and that SQL search seemed to be the only way to achieve it.
